### ISSUE TITLE
Add text input component

### ISSUE THREAD

**Issue by AaronCGoidel** [AaronCGoidel/CLImate#9]
Create an interaction component which prompts the user and then takes input

**Comment by AdnCodez** [comment]
I'm going to start working on this issue, is there anything that I should keep in mind?

**Comment by AaronCGoidel** [comment]
I think that the underlying generic for the text input should allow for some flexibility in terms of what can and will be built on top of it. For example, this will probably be the basis for numeric input, so there needs to be the option to add input validation. Just keep in mind that the base component should be as customizable as possible.

**Comment by AdnCodez** [comment]
Okay, I got it;
I read the code you have and understand the way you structure things it is flexible, to say the least.
I thought about validation may be using regex or someway when the input is expected to be an email, for example, I think I will start with these basic things and then build on top of it (see attachment)
![Screenshot from 2019-10-09 18-40-06](https://user-images.githubusercontent.com/20237313/66505942-41351c00-eac4-11e9-9fdb-242e5c6a5297.png)

**Comment by AaronCGoidel** [comment]
Sounds good to me, checking if a string matches some format is going to be important. Something else I thought of which may be good to have is interchangeable prompting strings. Here's what I mean by that.

```
Enter your name:
> 
```
```
What is your name?
Name: 
```
```
How much money?
$
```

There should be two different options, one for the prompt and one for string which starts the input.

**Comment by AdnCodez** [comment]
I see I'll try to implement it so that it is set as an argument, and make like a default one as > or ↣

**Comment by AaronCGoidel** [comment]
Yeah. For now, I'd go with > as the default. In the future my idea is to implement profiles of some sort which will change the default characters. I'd like them to be as consistent as possible throughout the package. For example, every arrow or pointer should be the same character.
